clang/lib/Driver/Driver.cpp

```diff
@@ -6754,14 +6754,28 @@ const char *Driver::GetNamedOutputPath(Compilation &C, const JobAction &JA,
   }
 
   // Redirect output for the generated source + integration footer.
-  if (isa<AppendFooterJobAction>(JA) && !isSaveTempsEnabled()) {
+  if (isa<AppendFooterJobAction>(JA)) {
     if (Arg *A = C.getArgs().getLastArg(options::OPT_fsycl_footer_path_EQ)) {
       SmallString<128> OutName(A->getValue());
       StringRef BaseName = llvm::sys::path::filename(BaseInput);
-      std::string TmpName =
-          GetTemporaryPath(llvm::sys::path::stem(BaseName),
-                           types::getTypeTempSuffix(JA.getType()));
-      llvm::sys::path::append(OutName, llvm::sys::path::filename(TmpName));
+      std::string TmpName;
```

Comment:
This can be declared + initialized inside the succeeding `else` block.

Reply:
Wasn't sure due to scoping.  `TmpName` is populated in the else but is used outside of it.